airbyte-webapp/src/packages/cloud/services/thirdParty/fullstory/useFullStory.tsx

```diff
@@ -18,6 +20,23 @@ const useFullStory = (
     }
   }, [config]);
 
+  useEffect(() => {
+    if (enabled) {
+      if (user) {
```

Comment:
What happens if Fullstory was not inited yet? Should we call identify only if Fullstory was already inited?

Reply:
As long as we call `FullStory.init()` before `FullStory.identify()` it shouldn't be a problem, since `init` is sync and we're fine to call any of the other methods directly after it. Since `useEffects` are executed in order and there is no combination of flags that would cause the first effect hoook not to run the `init` method, but the second to run the `identity` call, I think this should be fine (and also didn't cause any errors during testing).

Reply:
Hmmm, but as Fullstory script is loaded in async 

```
o=n.createElement(t);o.async=1;o.crossOrigin='anonymous';o.src='https://'+_fs_script;
```
it still makes me feel that it is still possible to face such a situation, though it could be very rare.

Probably we don't need to spend time on it now and only change behavior if we notice in logs any Fullstory warnings or errors.

Reply:
I see what you mean. The actual full script is loaded async, but that's not a problem, since the fullstory snippet will immediately when `init` is called, create the API functions (like `identity`), which will write in a queue, from which that script is loaded. See https://github.com/fullstorydev/fullstory-browser-sdk/blob/main/src/snippet.js#L23, `g.q` is the queue in which all events will be stored until the actual async loaded script is loaded and reads that queue. This is a common pattern for snippet functions like this, so you don't need to anyhow wait for async loaded scripts. Our used version of that library looks a bit different than the linked code, but has the same queue behavior.

Reply:
Indeed! Good catch about the queue. Then we won't have any problems.